pkg/util/validation/validation.go

```diff
@@ -29,6 +32,18 @@ func ValidatePropagationSpec(spec policyv1alpha1.PropagationSpec) field.ErrorLis
 	return allErrs
 }
 
+// ValidateResourceSelectors validates all resourceSelectors before creation or update.
+func ValidateResourceSelectors(resourceSelectors []policyv1alpha1.ResourceSelector, fldPath *field.Path) field.ErrorList {
```

Comment:
For cpp, we need to validate the namespace together with name if it matches a namespace-scoped resource.

Reply:
It has a certain impact on performance because we need to query `APIResourceList` every time we verify, do we have to do this?

Reply:
when CRD is not installed, how to check if it is a namespaced resource? people can choose to create propagation policies first, then install CRDs and propagate resources when needed.
cc @RainbowMango for comments.

Reply:
> I think there is not much diff in performance because if it's a namespace-scope resource, you need to traverse a namespace in the end if you don't fill in.

I didn't get it. Maybe @Poor12 can explain more.

Reply:
@Poor12 means that if the namespace of `ResourceSelector` unspecified for `CPP`, namespaced resource with the same name under different namespaces may be preempted during policy preemption. If I am wrong, please @Poor12 correct me.

> when CRD is not installed, how to check if it is a namespaced resource? people can choose to create propagation policies first, then install CRDs and propagate resources when needed.

But I think there are still some problems with doing this, I don't know if it‘s necessary to verify.